src/lib/reasoners/arrays_rel.ml

```diff
@@ -85,91 +87,91 @@ let timer = Timers.M_Arrays
 
 module H = Ephemeron.K1.Make (Expr)
 
-type t =
-  {gets  : G.t;               (* l'ensemble des "get" croises*)
-   tbset : S.t TBS.t ;        (* map t |-> set(t,-,-) *)
-   split : LRset.t;           (* l'ensemble des case-split possibles *)
-   conseq   : Conseq.t LRmap.t; (* consequences des splits *)
-   seen  : E.Set.t Tmap.t;    (* combinaisons (get,set) deja splitees *)
-   new_terms : E.Set.t;
-   size_splits : Numbers.Q.t;
-   cached_relevant_terms : (G.t * S.t TBS.t) H.t;
-  }
-
-
-let empty uf =
-  {gets  = G.empty;
-   tbset = TBS.empty;
-   split = LRset.empty;
-   conseq   = LRmap.empty;
-   seen  = Tmap.empty;
-   new_terms = E.Set.empty;
-   size_splits = Numbers.Q.one;
-   cached_relevant_terms = H.create 1024;
-  }, Uf.domains uf
+(* A read is a term of the form `(select a i)` for some array `a` and index `i`
+   and a write is a term of the form `(store a i v)`.
 
-(*BISECT-IGNORE-BEGIN*)
-module Debug = struct
-  open Printer
+   A relevant index is a term that appears as an index in read/write terms.
 
-  let assume la =
-    let print fmt (a,_,_,_) =
-      Format.fprintf fmt "> %a@,"
-        LR.print (LR.make a)
-    in
-    if Options.get_debug_arrays () && la != [] then
-      Printer.print_dbg
-        ~module_name:"Arrays_rel"
-        ~function_name:"assume"
-        "@[<v 2>We assume: @ %a" (pp_list_no_space print) la
-
-  (* unused --
-     let print_gets fmt = G.iter (fun t -> fprintf fmt "%a@." E.print t.g)
-     let print_sets fmt = S.iter (fun t -> fprintf fmt "%a@." E.print t.s)
-     let print_splits fmt =
-     LRset.iter (fun a -> fprintf fmt "%a@." LR.print a)
-     let print_tbs fmt =
-     TBS.iter (fun k v -> fprintf fmt "%a --> %a@." E.print k print_sets v)
-
-     let env fmt env =
-     if get_debug_arrays () then begin
-      fprintf fmt "-- gets ----------------------------------------@.";
-      print_gets fmt env.gets;
-      fprintf fmt "-- tabs of sets --------------------------------@.";
-      print_tbs fmt env.tbset;
-      fprintf fmt "-- splits --------------------------------------@.";
-      print_splits fmt env.split;
-      fprintf fmt "------------------------------------------------@."
-     end
-  *)
-
-  let new_equalities st =
-    if Options.get_debug_arrays () then begin
-      Printer.print_dbg
-        ~module_name:"Arrays_rel"
-        ~function_name:"new_equalities"
-        "@[<v 2>%d implied equalities"
-        (Conseq.cardinal st);
-      Conseq.iter (fun (a,ex) ->
-          Printer.print_dbg ~header:false
-            "%a : %a" E.print a Ex.print ex
-        ) st
-    end
+   We say that a read/write is known if we have seen this term in [assume]. *)
+type t = {
+  gets : G.t;
+  (* Set of all the known reads. *)
```

Comment:
The documentations talks about reads and writes (referring to `select` and `store`) but the code refers to them as `set`s and `get`s. I think it should be harmonized and only one notation should be used.

Maybe at some point we should simply switch to the standard and use `select` and `store` internally, but I think `select` is used in the `ADT` theory as well.

Reply:
I agree, it's a mess... There are three notations:
- `set/get` from the code
- `select/store` from the SMT-LIB standard,
- `read/write` from many papers on Array theories.

In my opinion, we should use the SMT-LIB standard but, as you pointed out, `select` is already in use in the ADT theory. I suggest that we keep `set/get` in Array and harmonize notations later.

Reply:
I've pushed a new commit to address your comments. Is it better?

Reply:
Yes! LGTM